### DISCUSSION TITLE
"procedure entry point" Error?

### DISCUSSION THREAD

**Discussion by chizZo1** [GyulyVGC/sniffnet#212] [Q&A]
Hello!
I try to get the App to work, installed the dependencies, downloaded the Npcap SDK and added the SDK's /Lib/x64 folder to my LIB environment variable.
one error is gone now, but now i get this:

"The procedure entry point 'pcap_set_immediate_mode' can not be found in the DLL 'C:\Program Files\sniffnet\bin\sniffnet.exe'"

Any ideas?

Thank you!

**Comment by GyulyVGC** [comment]
Hi!
Have you set `Install Npcap in WinPcap API-Compatible Mode` during installation of Npcap?

**Comment by chizZo1** [comment]
now i am not so sure anymore... isnt it pre selected? if not i read something wrong...  then just uninstall and reinstall?

thank you for the fast response!

**Comment by GyulyVGC** [comment]
Not always apparently, since this issue has already been documented for example in #201 and #211.

Let me know if reinstalling it with that option fixes the problem for you as well.

**Comment by chizZo1** [comment]
oh shit, sorry...!

okay, i gonna try and let you know!

**Comment by GyulyVGC** [comment]
Don't worry!

**Comment by chizZo1** [comment]
okay, installed it again, but it was pre selected... doesnt seem to be the problem
does it matter where i put the npcap sdk?
C:\Users\me1234\Downloads\npcap-sdk-1.13\Lib\x64

**Comment by GyulyVGC** [comment]
Nope, that's not a problem.

The problem is most likely due to Npcap version. Which version did you install?
Different people solved the problem simply by changing their Npcap to a different version.

For example, from #86:
> FWIW, I had the same issue.
> 
> I already had Npcap 1.71 installed. I downloaded the Npcap SDK 1.13 and set `%LIB%` to the `npcap-sdk-1.13\Lib\x64` folder. Didn't help. I also tried adding `Program Files\Npcap` to `%PATH%`, that didn't help either.
> 
> Eventually I installed Npcap 1.74 and 🎉 now it works.

**Comment by chizZo1** [comment]
Npcap 1.75 installer and Npcap SDK 1.13 

the name of the variable is LIB or %LIB%?

**Reply by GyulyVGC** [reply]
> Npcap 1.75 installer and Npcap SDK 1.13
> 
> the name of the variable is LIB or %LIB%?

It's correct as you put, LIB.

I can suggest you to downgrade to [Npcap 1.74](https://npcap.com/dist/npcap-1.74.exe), hoping this will be useful for you

**Comment by chizZo1** [comment]
okay, i gonna try that and let u know if it works :)

thanks for the help!

**Comment by GyulyVGC** [comment]
Don't worry, it' my pleasure.
Also, I'd like to package Sniffnet Installer with this installation process made automatically, so it' very useful for me to see all the various problem to pick the best version/configuration for the next releases.

**Comment by chizZo1** [comment]
okay :)

one last, maybe stupid question.. is it nessesary to restart the system after deinstall reinstall and so on?

**Comment by GyulyVGC** [comment]
It shouldn't be necessary. If it doesn't work even with this version I can link you an older version which solved the problem for other guys. As a last resort, if nothing works, you can try restarting but usually it's not needed.

**Comment by chizZo1** [comment]
okay, thanks again :)

**Comment by chizZo1** [comment]
same issue with 1.74... didnt had time to restart yet, gonna do later... but also gonna try on another pc...
the one i currently try is a nuc12 pro wit i5 and windows 11... also very fresh installation, maybe something else missing?

**Comment by GyulyVGC** [comment]
If you correctly set the environment variable you should be fine in theory.

This is [Npcap 1.60](https://npcap.com/dist/npcap-1.60.exe) but I doubt it will solve the problem at this point.

Let me know if you have updates.

**Comment by chizZo1** [comment]
looks like this:
![Screenshot 2023-05-22 102057](https://github.com/GyulyVGC/sniffnet/assets/134257796/e44562f5-35f2-4d54-a3ed-5200092cddf5)

**Comment by GyulyVGC** [comment]
It's correct.

**Comment by chizZo1** [comment]
little update, on my gaming pc it works just fine.. installed 1.75 there..

on my nuc still same error... any more ideas?
the nuc is actually running 24/7, so would be nice to get the app working on that machine :)

**Comment by GyulyVGC** [comment]
I'm happy it works on your gaming PC at least.

Did you already have any other version of wpcap or npcap installed?

Your same problem has also been solved by deleting them in #54

**Comment by chizZo1** [comment]
oh, no, last i tried was npcap 1.74...  okay, i try the other version u linked me and follow the other post too... give u an update later :)

**Comment by GyulyVGC** [comment]
If nothing works, try also Npcap 1.60 linked above.
